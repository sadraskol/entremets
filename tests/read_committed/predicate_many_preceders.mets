init do
    insert into users (id, age) values (1, 10)
end

process do
    begin read_committed
    let first := select age from users where age = 30
latch
latch
    let second := select age from users where age % 3 = 0
    commit
end

process do
    begin read_committed
latch
    insert into users(id, age) values (3, 30)
    commit
end

property = eventually<first = {} and second = {}>
