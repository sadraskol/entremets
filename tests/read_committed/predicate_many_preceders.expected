Following property was violated: eventually<first = {} and second = {}>